mongo/datadog_checks/mongo/dbm/operation_samples.py

```diff
@@ -0,0 +1,347 @@
+            "collection": self._get_command_collection(command),
+            "comment": command.get("comment"),
+            "truncated": self._get_command_truncation_state(command),
+            "client": self._get_operation_client(operation),
+            "user": self._get_operation_user(operation),
+        }
+
+    def _get_operation_stats(self, operation: dict) -> OperationSampleOperationStats:
+        return {
+            "active": operation.get("active", False),  # bool
+            "desc": operation.get("desc"),  # str
+            "opid": operation.get("opid"),  # str
+            "ns": operation.get("ns"),  # str
+            "plan_summary": operation.get("planSummary"),  # str
+            "current_op_time": operation.get("currentOpTime"),  # str  start time of the operation
+            "microsecs_running": operation.get("microsecs_running"),  # int
+            "transaction_time_open_micros": operation.get("transaction", {}).get("timeOpenMicros"),  # int
+            # Conflicts
+            "prepare_read_conflicts": operation.get("prepareReadConflicts", 0),  # int
+            "write_conflicts": operation.get("writeConflicts", 0),  # int
+            "num_yields": operation.get("numYields", 0),  # int
+            # Locks
+            "waiting_for_lock": operation.get("waitingForLock", False),  # bool
+            "locks": self._format_key_name(operation.get("locks", {})),  # dict
+            "lock_stats": self._format_key_name(operation.get("lockStats", {})),  # dict
+            # Flow control
+            "waiting_for_flow_control": operation.get("waitingForFlowControl", False),  # bool
+            "flow_control_stats": self._format_key_name(operation.get("flowControlStats", {})),  # dict
+            # Latches
+            "waiting_for_latch": operation.get("waitingForLatch", False),  # bool
+        }
+
+    def _get_query_signature(self, obfuscated_command: str) -> str:
+        return compute_exec_plan_signature(obfuscated_command)
+
+    def _create_operation_sample_payload(
+        self,
+        now: float,
+        operation_metadata: OperationSampleOperationMetadata,
+        obfuscated_command: str,
+        query_signature: str,
+        explain_plan: OperationSamplePlan,
+        activity: OperationSampleActivityRecord,
+    ) -> OperationSampleEvent:
+        event = {
+            "host": self._check._resolved_hostname,
+            "dbm_type": "plan",
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "mongodb",
+            "ddtags": ",".join(self._check._get_tags(include_deployment_tags=True)),
+            "timestamp": now * 1000,
+            "network": {
+                "client": operation_metadata["client"],
+            },
+            "db": {
+                "instance": operation_metadata["dbname"],
+                "plan": explain_plan,
+                "query_signature": query_signature,
+                "resource_hash": query_signature,
```

Comment:
what is resource_hash for?

Reply:
originally intended for apm link but not used for now. will remove